classes/helper.php

```diff
@@ -0,0 +1,127 @@
+ */
+class helper {
+
+    /**
+     * Get an installed library.
+     *
+     * @param array $options Array containing either id or a title and version number.
+     * @return \stdClass Record from hvp_libraries.
+     */
+    public static function get_library(array $options): \stdClass {
+        global $DB;
+        if (!empty($options['library-id'])) {
+            $libraries = $DB->get_records('hvp_libraries', ['id' => $options['library-id']]);
+        } else if (!empty($options['library-title']) && !empty($options['library-version'])) {
+            $versionparts = explode('.', $options['library-version']);
+            if (empty($versionparts) || count($versionparts) !== 3) {
+                throw new \moodle_exception('error:invalidversionformat', 'hvp');
+            }
+            $libraries = $DB->get_records('hvp_libraries', [
+                'title' => $options['library-title'],
+                'major_version' => $versionparts[0],
+                'minor_version' => $versionparts[1],
+                'patch_version' => $versionparts[2],
+            ]);
+        } else if (!empty($options['library-title'])
+                && !empty($options['major-version'])
+                && !empty($options['minor-version'])
+                && !empty($options['patch-version'])) {
+            $libraries = $DB->get_records('hvp_libraries', [
+                'title' => $options['library-title'],
+                'major_version' => $options['major-version'],
+                'minor_version' => $options['minor-version'],
+                'patch_version' => $options['patch-version'],
+            ]);
+        }
+
+        if (empty($libraries)) {
+            throw new \moodle_exception('error:nolibraryexists', 'hvp');
+        }
+
+        if (count($libraries) !== 1) {
+            throw new \moodle_exception('error:multiplelibrariesfound', 'hvp');
+        }
+
+        return reset($libraries);
+    }
+
+    /**
+     * Get list of libraries dependant on provided library.
+     *
+     * @param \stdClass $library An hvp_library record.
+     * @return array List of titles of libraries depending on provided library.
+     */
+    public static function get_dependant_libraries(\stdClass $library): array {
+        global $DB;
+        $titles = [];
+        $framework = \mod_hvp\framework::instance('interface');
+        list('content' => $content, 'libraries' => $dependentlibraries) = $framework->getLibraryUsage($library->id);
+
+        if (empty($dependentlibraries)) {
+            return $titles;
+        }
+
+        // If dependencies are found, get a list of titles with version number.
+        $dependencies = $DB->get_records('hvp_libraries_libraries', ['required_library_id' => $library->id],
+            '', 'library_id');
+        $dependencies = array_keys($dependencies);
+        list($insql, $params) = $DB->get_in_or_equal($dependencies);
+        $sql = "SELECT id, title, major_version, minor_version, patch_version
```

Comment:
Can we format this to be use the standard Moodle SQL formatting

Reply:
Yep, good pick.

Reply:
Done